iron-a11y-keys-behavior.html

```diff
@@ -0,0 +1,335 @@
+          validKey = String(96 - keyCode);
+        } else {
+          validKey = KEY_CODE[keyCode];
+        }
+      }
+      return validKey;
+    }
+
+    function normalizedKeyForEvent(keyEvent) {
+      // fall back from .key, to .keyIdentifier, to .keyCode, and then to
+      // .detail.key to support artificial keyboard events
+      return transformKey(keyEvent.key) ||
+        transformKeyIdentifier(keyEvent.keyIdentifier) ||
+        transformKeyCode(keyEvent.keyCode) ||
+        transformKey(keyEvent.detail.key) || '';
+    }
+
+    function keyComboMatchesEvent(keyCombo, keyEvent) {
+      return normalizedKeyForEvent(keyEvent) === keyCombo.key &&
+        keyEvent.shiftKey === !!keyCombo.shiftKey &&
+        keyEvent.ctrlKey === !!keyCombo.ctrlKey &&
+        keyEvent.altKey === !!keyCombo.altKey &&
+        keyEvent.metaKey === !!keyCombo.metaKey;
+    }
+
+    function parseKeyComboString(keyComboString) {
+      return keyComboString.split('+').reduce(function(parsedKeyCombo, keyComboPart) {
+        var eventParts = keyComboPart.split(':');
+        var keyName = eventParts[0];
+        var event = eventParts[1];
+
+        if (keyName in MODIFIER_KEYS) {
+          parsedKeyCombo[MODIFIER_KEYS[keyName]] = true;
+        } else {
+          parsedKeyCombo.key = keyName;
+          parsedKeyCombo.event = event || 'keydown';
+        }
+
+        return parsedKeyCombo;
+      }, {
+        combo: keyComboString.split(':').shift()
+      });
+    }
+
+    Polymer.IronA11yKeysBehavior = {
+      properties: {
+        /**
+         * The HTMLElement that will be firing relevant KeyboardEvents.
+         */
+        keyEventTarget: {
+          type: Object,
+          value: function() {
+            return this;
+          }
+        },
+
+        _boundKeyHandlers: {
+          value: function() {
+            return [];
+          }
+        }
+      },
+
+      observers: [
+        '_resetKeyBindings(keyEventTarget, _boundKeyHandlers)'
+      ],
+
+      keyBindings: {},
+
+      registered: function() {
+        this._prepKeyBindings();
+      },
+
+      addOwnKeyBinding: function(eventString, handlerName) {
+        if (!this.hasOwnProperty('_keyBindings')) {
+          this._keyBindings = Object.create(this._keyBindings || {});
+        }
+
+        this._addKeyBinding(eventString, handlerName);
+        this._resetKeyBindings();
+      },
+
+      _addKeyBinding: function(eventString, handlerName) {
+        eventString.split(' ').forEach(function(keyComboString) {
+          var keyCombo = parseKeyComboString(keyComboString);
+
+          this._keyBindings[keyCombo.event] =
+            this._keyBindings[keyCombo.event] || [];
+
+          this._keyBindings[keyCombo.event].push([
+            keyCombo,
+            handlerName
+          ]);
+        }, this);
+      },
+
+      _prepKeyBindings: function() {
+        this._keyBindings = {};
+
+        this.behaviors.concat(this).forEach(function(node) {
```

Comment:
I see, this is for if multiple behaviors want to use key bindings. Since this is called at register time, does it handle per instance `keyBindings` (or not supported)?